source/java/spring-boot-web/tutorial.html

```diff
@@ -0,0 +1,845 @@
+<span class="nv">STORMPATH_API_KEY_FILE</span><span class="o">=</span>&lt;path to your key file&gt; <span class="se">\</span>
+<span class="nv">STORMPATH_AUTHORIZED_GROUP_USER</span><span class="o">=</span>&lt;href to your group in Stormpath&gt; <span class="se">\</span>
+java -jar build/libs/02-spring-security-ftw-0.1.0.jar
+</pre></div>
+</div>
+<p>In the next section, we&#8217;ll add a small amount of code to be able to dynamically set the Group reference and make the code more readable.</p>
+</div>
+<div class="section" id="spring-security-refined">
+<span id="id9"></span><h2><a class="toc-backref" href="#id21">Spring Security Refined</a><a class="headerlink" href="#spring-security-refined" title="Permalink to this headline">¶</a></h2>
+<p>The code for this section can be found <a class="reference external" href="https://github.com/stormpath/stormpath-sdk-java/tree/master/tutorials/spring-boot-default/03-spring-security-refined">here</a>.</p>
+<p>In the previouse section, we hard-coded the Stormpath group href into <code class="docutils literal"><span class="pre">HelloService</span></code>.</p>
+<p>This is cumbersome in a real world situation where you may have multiple environments (dev, stage, prod, etc.).
+You don&#8217;t want to have to change source, recompile and deploy for a new environment or when you change a group in Stormpath.</p>
+<p>With a little bit of Spring magic and Stormpath&#8217;s super flexible configuration mechanism, we can make this much more dynamic.</p>
+<p>Take a look at the updated <code class="docutils literal"><span class="pre">HelloService</span></code> class:</p>
+<div class="highlight-java"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
+2
+3
+4
+5
+6
+7
+8</pre></div></td><td class="code"><div class="highlight"><pre><span class="nd">@Service</span>
+<span class="kd">public</span> <span class="kd">class</span> <span class="nc">HelloService</span> <span class="o">{</span>
+<span class="hll">    <span class="nd">@PreAuthorize</span><span class="o">(</span><span class="s">&quot;hasRole(@Roles.USER)&quot;</span><span class="o">)</span>
+</span>    <span class="kd">public</span> <span class="n">String</span> <span class="nf">sayHello</span><span class="o">(</span><span class="n">Account</span> <span class="n">account</span><span class="o">)</span> <span class="o">{</span>
+        <span class="k">return</span> <span class="s">&quot;Hello, &quot;</span> <span class="o">+</span> <span class="n">account</span><span class="o">.</span><span class="na">getGivenName</span><span class="o">()</span> <span class="o">+</span>
+            <span class="s">&quot;. You have the required permissions to access this restricted resource.&quot;</span><span class="o">;</span>
+    <span class="o">}</span>
+<span class="o">}</span>
+</pre></div>
+</td></tr></table></div>
+<p>Spring has an expression language called SpringEL, that let&#8217;s you reference objects and properties found in your Spring
```

Comment:
Please add a link to http://docs.spring.io/spring/docs/current/spring-framework-reference/html/expressions.html